File: framework/java/src/main/java/soda/unittest/work/parse/CharList2dConverter.java
package soda.unittest.work.parse;

import java.util.List;
import java.util.stream.Collectors;

public class CharList2dConverter implements ObjectConverter<List<List<Character>>, List<List<String>>> {

    private CharListConverter conv = new CharListConverter();

    @Override
    public List<List<Character>> fromJsonSerializable(List<List<String>> lists) {
        return lists.stream().map(s -> conv.fromJsonSerializable(s)).collect(Collectors.toList());
    }

    @Override
    public List<List<String>> toJsonSerializable(List<List<Character>> lists) {
        return lists.stream().map(c -> conv.toJsonSerializable(c)).collect(Collectors.toList());
    }
}

File: framework/scala/src/main/scala/soda/scala/unittest/CharList2dConverter.scala
package soda.scala.unittest
import play.api.libs.json.{JsValue, Json}

class CharList2dConverter extends ObjectConverter[List[List[Char]]] {

  override def fromJsonSerializable(js: JsValue): List[List[Char]] = {
    js.as[List[List[String]]].map(CharListConverter.s2c)
  }

  override def toJsonSerializable(mx: List[List[Char]]): JsValue = {
    Json.toJson(mx.map(CharListConverter.c2s))
  }
}
